File: JavaScript/condition.js
//สร้าง function เพื่อคำนวณเกรด โดยใช้ if else-if และ else
function gradeCalculate(score) { // รับ parameter เป็นคะแนน(number)
    let grade;
    if (score >= 80) { //ใช้ if เงื่อนไขแรก
        grade = 'A';
    } else if (score >= 70 && score < 80) { // ใช้ else if ถ้ายังมีเงื่อนไขต่อไปเรื่อยๆ
        grade = 'B';
    } else { //ใช้ else เมื่อเป็นเงื่อนไขสุดท้าย
        grade = 'F';
    }
    return grade;
}

// function นี้ใช้โชว์เกรดออกมาเป็นข้อความ โดยใช้ switch case
function showGrade(grade) {
    let message;
    switch (grade) {
        case 'A':
            message = "Your grade is A";
            break;
        case 'B':
            message = "Your grade is B";
            break;
        case 'F':
            message = "Your grade is F"
            break;
        default:
            message = "Invalid grade";
            break;
    }

    console.log(message);
}

function main(){
    let studen1 = gradeCalculate(62); // ใส่คะแนนเพื่อคิดเกรด
    showGrade(studen1); //รับเกรดจาก score1 เพื่อแสดงผล
}

main(); // run code ใน function main

File: Clojure/condition.clj
; gradeCalculate รับคะแนนและคำนวณออกมาเป็นเกรด โดยใช้ if
; ในภาษา clojure จะไม่มี if, else-if แต่จะใช้เป็น if ไปเรื่อยๆ และถ้าเป็นเงื่อนไขสุดท้ายจะเขียนเฉพาะคำสั่ง เช่น "F"
(defn gradeCalculate [score]
  (if (>= score 80) ; เงื่อนไขแรก
    "A" ; if true, grade = A, false ไปเงื่อนไขต่อไป
    (if (and (>= score 70) (< score 80)) ; เงื่อนไขที่ 2
      "B" ; if true, grade = B, false ไปเงื่อนไขสุดท้าย
      "F"))) ; คล้ายกับ else ในภาษาอื่นๆ ในกรณีที่ไม่เข้าเงื่อนไขใดๆ grade = F

; showGrade แสดงผลเกรดออกมาเป็นข้อความ โดยใช้ switch case
(defn showGrade [grade] ; รับ grade ที่ส่งมาจาก function gradeCalculate
  (let [message (case grade ; ใช้ switch case เพื่อเช็คเกรดว่าตรงกับเคสใด และทำการแสดงผลตามเกรด
                  "A" "Your grade is A"
                  "B" "Your grade is B"
                  "F" "Your grade is F"
                  "Invalid grade")]
    (println message)))

; เรียกใช้งาน function ใน main
(defn main []
  (let [score (gradeCalculate 72)] ;ใส่คะแนน =  72
    (showGrade score)))

; run function main
(main)
